cpp/src/Slice/SliceUtil.cpp

```diff
@@ -509,8 +509,7 @@ Slice::isProxyType(const TypePtr& type)
 bool
 Slice::isFirstElement(const ContainedPtr& p)
 {
-    ContainerPtr container = dynamic_pointer_cast<Container>(p->container());
-    return p == container->contents().front();
```

Comment:
Unrelated to this PR. But `container()` returns a `ContainerPtr`... so... no idea why we're dynamic casting here.